### ISSUE TITLE
<script setup> vue/no-mutating-props  - Unexpected mutation of "isExpanded" prop

### ISSUE THREAD

**Issue by soerenmartius** [vuejs/eslint-plugin-vue#1702]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7.32.0
- **eslint-plugin-vue version:** 8.0.3
- **Node version:** 14.17.2
- **Operating System:** Darwin / MacOS

**What did you do?**	

```vue
<script setup lang="ts">
import { defineEmits, defineProps, ref, watch, withDefaults } from 'vue'

import { FormEvents } from './types'

const emit = defineEmits<{
  (e: FormEvents.TOGGLED, isExpanded: boolean): void
}>()

interface Props {
  isExpanded?: boolean
}

const props = withDefaults(defineProps<Props>(), {
  isExpanded: false,
})

const isExpanded = ref(props.isExpanded)

watch(
  () => props.isExpanded,
  () => {
    isExpanded.value = props.isExpanded
  },
)

watch(isExpanded, () => emit(FormEvents.TOGGLED, isExpanded.value))
</script>
```

**What did you expect to happen?**

As `isExpanded` is redefined as a `ref`, eslint should pass. Instead it detects an error:

![Screenshot 2021-11-10 at 14 28 07](https://user-images.githubusercontent.com/930643/141121704-f74c7036-919d-40c0-baac-1315a0812400.png)

**What actually happened?**

Eslint should pass since the prop doesn't get mutate directly.

**Comment by ota-meshi** [comment]
I can't reproduce your problem. Please provide a repository to reproduce the problem.

**Comment by soerenmartius** [comment]
> I can't reproduce your problem. Please provide a repository to reproduce the problem.

Hey @ota-meshi , thanks for coming back to me!

Of course, I just uploaded an example to reproduce the error here: https://github.com/soerenmartius/vitesse/blob/soerenmartius/eslint-test/src/pages/index.vue

**Comment by ota-meshi** [comment]
I was able to reproduce the issue.
Need a template block to reproduce issue.

```vue
<template>
  <input v-model="isExpanded">
</template>

<script setup lang="ts">
import { defineEmits, defineProps, ref, watch, withDefaults } from 'vue'

import { FormEvents } from './types'

const emit = defineEmits<{
  (e: FormEvents.TOGGLED, isExpanded: boolean): void
}>()

interface Props {
  isExpanded?: boolean
}

const props = withDefaults(defineProps<Props>(), {
  isExpanded: false,
})

const isExpanded = ref(props.isExpanded)

watch(
  () => props.isExpanded,
  () => {
    isExpanded.value = props.isExpanded
  },
)

watch(isExpanded, () => emit(FormEvents.TOGGLED, isExpanded.value))
</script>
```

[Demo](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx9Uz1v4zAM/SuEltqAnAAdDSfXoWmXAr3hxiw6m05V6AsS7esh8H8/WgliN8Mttun3+B5JUWfR+g5FLRpCG4wi3B8dQKNdGAjGyjJqdkeh0+ErKNdhdxTMaLYrOoepjToQJKQhgFHuxCmUMlXb4CPBGTrstcOD1ZTkNfgZfeAgYi/hj6L2g1+aPp6xV4OhBBP00Vt4GAd8mH1uWi8+2sOIbsXZbOlvwJR5rXeJANkKdmvf5jw3V2C9Etj8en99fTs8S1h6rOG39waVK2sYve6ObtoXZa7AEcZetQi5dsiCS+KPWyanLJWEzN19a65YjaDJT7aQ94I19MoklCxXLnoLzqI8vSIbbJbfmZsnWuSOS9jt4Z4kFyi7rn03ozIDsvp90kycOHPlsFK8ys2TL/4/4otBWc67dNmevZAiDgaTqM+CT3zrfGUHUqTdqcpliPpRZqT11rJw1emILekRb8hn+qqGhKkaVVz4F4Mqr+c3eJIi8BdGvgFPeX8uTEyGT3p7xaZ/7J8cZw==)
